File: daily/Arrays/795.Number-of-subarrays-with-bounded-maximum.go
package daily

func numSubarrayBoundedMax(nums []int, left int, right int) int {
	return count(nums, right) - count(nums, left-1)
}

func count(nums []int, bound int) int {
	cur := 0
	total := 0
	for _, num := range nums {
		if num <= bound {
			cur += 1
		} else {
			cur = 0
		}
		total += cur
	}
	return total
}

File: daily/Arrays/795.Number-of-subarrays-with-bounded-maximum.py
from typing import List


class Solution:
    def numSubarrayBoundedMax(self, nums: List[int], left: int, right: int) -> int:
        return self._count(nums, right) - self._count(nums, left - 1)

    def _count(self, nums: List[int], bound: int) -> int:
        cur = 0
        total = 0
        for x in nums:
            if x <= bound:
                cur += 1
            else:
                cur = 0
            total += cur
        return total

